sensor/common/clusterentities/store.go

```diff
@@ -58,23 +58,11 @@ func (ed *EntityData) String() string {
 }
 
 // GetDetails returns the internal data about the entity
-func (ed *EntityData) GetDetails() (containerID, podIP string) {
-	for s := range ed.containerIDs {
-		containerID = s
-		break
-	}
-	// Sort the IPs to deterministically return the first IP if more than one are stored.
-	// The slice should contain single element in most of the cases.
-	sortedIPs := slices.SortedFunc(maps.Keys(ed.ips), func(a, b net.IPAddress) int {
-		return strings.Compare(a.String(), b.String())
+func (ed *EntityData) GetDetails() (containerIDs []string, podIPs []net.IPAddress) {
+	containerIDs = slices.Sorted(maps.Keys(ed.containerIDs))
```

Comment:
nit: would it make sense to do the sorting and filtering in updateHeritageData()? That way, the comment `// Deliberately choosing only the first IP from potentially many` would be easier to understand without coming here to see that the data is sorted.

On the other hand it is also nice to get consistent results, and perhaps it also makes sense to serve this data sorted to other consumers.

From my point of view it's fine to do it either way :)